numba/tests/test_stencils.py

```diff
@@ -365,12 +407,1272 @@ def test_impl(n):
+        self.check(kernel, a)
+
+    def test_basic14b(self):
+        """rel index add domain change const"""
+        a = np.arange(12).reshape(3, 4)
+
+        def kernel(a):
+            t = 1.j
+            return a[0, 0] + t
+        self.check(kernel, a)
+
+    def test_basic15(self):
+        """two rel index, add const"""
+        a = np.arange(12).reshape(3, 4)
+
+        def kernel(a):
+            return a[0, 0] + a[1, 0] + 1.
+        self.check(kernel, a)
+
+    # FIXME: pyStencil fails, IndexError
+    def test_basic16(self):
+        """two rel index OOB, add const"""
+        a = np.arange(12).reshape(3, 4)
+
+        def kernel(a):
+            return a[0, 0] + a[10, 0] + 1.
+        self.check(kernel, a)
+
+    def test_basic17(self):
+        """two rel index boundary test, add const"""
+        a = np.arange(12).reshape(3, 4)
+
+        def kernel(a):
+            return a[0, 0] + a[2, 0] + 1.
+        self.check(kernel, a)
+
+    def test_basic18(self):
+        """two rel index boundary test, add const"""
+        a = np.arange(12).reshape(3, 4)
+
+        def kernel(a):
+            return a[0, 0] + a[-2, 0] + 1.
+        self.check(kernel, a)
+
+    def test_basic19(self):
+        """two rel index boundary test, add const"""
+        a = np.arange(12).reshape(3, 4)
+
+        def kernel(a):
+            return a[0, 0] + a[0, 3] + 1.
+        self.check(kernel, a)
+
+    def test_basic20(self):
+        """two rel index boundary test, add const"""
+        a = np.arange(12).reshape(3, 4)
+
+        def kernel(a):
+            return a[0, 0] + a[0, -3] + 1.
+        self.check(kernel, a)
+
+    def test_basic21(self):
+        """same rel, add const"""
+        a = np.arange(12).reshape(3, 4)
+
+        def kernel(a):
+            return a[0, 0] + a[0, 0] + 1.
+        self.check(kernel, a)
+
+    def test_basic22(self):
+        """rel idx const expr folding, add const"""
+        a = np.arange(12.).reshape(3, 4)
+
+        def kernel(a):
+            return a[1 + 0, 0] + a[0, 0] + 1.
+        self.check(kernel, a)
+
+    def test_basic23(self):
+        """rel idx, work in body"""
+        a = np.arange(12.).reshape(3, 4)
+
+        def kernel(a):
+            x = np.sin(10 + a[2, 1])
+            return a[1 + 0, 0] + a[0, 0] + x
+        self.check(kernel, a)
+
+    def test_basic23a(self):
+        """rel idx, dead code should not impact rel idx"""
+        a = np.arange(12.).reshape(3, 4)
+
+        def kernel(a):
+            x = np.sin(10 + a[2, 1])
+            return a[1 + 0, 0] + a[0, 0]
+        self.check(kernel, a)
+
+    def test_basic24(self):
+        """1d idx on 2d arr"""
+        a = np.arange(12).reshape(3, 4)
+
+        def kernel(a):
+            return a[0] + 1.
+        self.check(kernel, a, expected_exception=[ValueError, TypingError])
+
+    # Should this work at all? does it make sense?
+    # pyStencil raises as there's no rel indices
+    def test_basic25(self):
```

Comment:
We need to decide if this makes sense, and document that it'll do things like this if so. I'm inclined to think the behaviour is a bit unexpected, I was surprised it worked.

Reply:
Changed it to raise and exception.  Done.